src/scanpy/_utils/__init__.py

```diff
@@ -713,6 +713,27 @@ def _(
     )
 
 
+@singledispatch
+def axis_nnz(X: np.ndarray, axis: Literal[0, 1]) -> np.ndarray:
```

Comment:
maybe `X: ArrayLike`? With the idea that this is the fallback branch that’s able to handle everything that numpy can convert into an array?